Rudimentary CMake support

torss opened the pull request:
Note that this changes `FastNoiseSIMD.h` to a CMake configured file (see line 42 in `CMakeLists.txt`).
So it might be a slight inconvenience if someone doesn't want to use CMake.
(I.e. perhaps you don't want to merge this.)

torss committed:
Add rudimentary CMake support

torss committed:
Add .cmake postfix to FastNoiseSIMD.h filename

Commit:
Add OpenSimplex2 alternative to Simplex

c0rp3n committed:
[CMake] Improve support and add find package support

c0rp3n committed:
[CMake] Add version and description and some cleanup

c0rp3n committed:
[CI] Add appveyor and travis support

c0rp3n committed:
[CMake] Didnt include cmake dependencies

c0rp3n committed:
Update README.md [skip ci]

c0rp3n committed:
[CMake] Fix linux build

c0rp3n committed:
Update README.md [skip ci]

c0rp3n committed:
[CMake] Properly set arch

c0rp3n committed:
Update README.md [skip ci]

c0rp3n committed:
Move FastNoiseSIMD_internal.h as it is not a public header

Commit:
Undo whitespace changes the IDE made without me noticing

c0rp3n committed:
Add base unit test

c0rp3n committed:
[CI] Travis CMake pass vcpkg flags

c0rp3n committed:
Add d postfix to libs and fix msvc build options

c0rp3n committed:
Remove config header

torss committed:
Merge pull request #1 from c0rp3n/master

Merge c0rp3n/fastnoise-simd to torss/FastNoiseSIMD

torss committed:
Merge branch 'pr/2'

torss committed:
Fix minor README.md typo

Auburns commented:
Thank you for your work on this, I'm still figuring out cmake. I'm using it in [FastNoise2](https://github.com/Auburns/FastNoise2) but I haven't hooked up any config or CI support.

Could you give a run down on what platform/config the CI builds cover and what setup will they need. 

Also what are the cmake options and setup steps, I will need to add some info to the wiki.

Thanks again for your help with this.

Auburns commented:
@torss Why has the open simplex change been included now?

torss committed:
Add FN_SET_c0rp3n_CXX_FLAGS CMake option

This was added because setting these flags for MSVC did cause
a runtime library mismatch in a project using FastNoiseSIMD
via add_subdirectory.

torss commented:
Oops, sorry, I completely forgot about this old PR from 2018. ;)
Feel free to close it if the changes are potentially annoying.

I'm currently only using my fork in a private test project. I just merged @c0rp3n's fork because I noticed that it apparently extends CMake support _(although I personally don't need that)_ and @KdotJPG's fork since I wanted to try out his `OpenSimplex2`. So @c0rp3n would have to tell you about the CI build part, I'm not familiar with that either.

BTW nice to see that you are working on a successor, didn't know about `FastNoise2` yet!

c0rp3n commented:
So with my fork I added support for `find_package` in CMake and config support, then it is just a basic CI setup building on both MSVC 2019 and GCC 7.4.

The config works by just calling `configure_file` in CMake it then replaces the `#cmakedefine` with `#define` when the option with the same name is enabled the new file with the replaced defines is then written to the include directory. See https://pastebin.com/Tk3CuGLY for an example output after the configure.

And for buiding it would just be
```bash
mkdir build
cd build
cmake -DFN_COMPILE_AVX2=ON -DFN_COMPILE_AVX512=OFF .. && make
```
and then to install
```bash
sudo make install
```
 With windows it differs slightly need to pass you need to set where you want to install it to on configure so
```bash
cmake -DCMAKE_INSTALL_PREFIX="E:/deps" ..
```
There will then be an install target added to the VS project which you can then use to install the library.

Edit:
You may also want to revert some of the readme changes as I just tried to make it less first person.
